tools/socktap/main.cpp

```diff
@@ -85,11 +88,36 @@ int main(int argc, const char** argv)
         mib.itsGnLocalGnAddr.mid(mac_address);
         mib.itsGnLocalGnAddr.is_manually_configured(true);
         mib.itsGnLocalAddrConfMethod = geonet::AddrConfMethod::MANAGED;
-        mib.itsGnSecurity = false;
 
         TimeTrigger trigger(io_service);
+
+        std::string securitySetting = vm["security"].as<std::string>();
+
+        // We always use the same ceritificate manager and crypto services for now.
+        // If itsGnSecurity is false, no signing will be performed, but receiving of signed messages works as expected.
+        auto certificate_manager_factory = security::builtin_certificate_managers();
+        auto certificate_manager = certificate_manager_factory.create("Naive", trigger.runtime());
+
+        auto crypto_backend_factory = security::builtin_backends();
+        auto crypto_backend = crypto_backend_factory.create();
```

Comment:
You can simplify these two lines:
`auto crypto_backend = security::create_backend("default");`

Reply:
Done. :+1: